addons/website/static/src/snippets/s_popup/options.js

```diff
@@ -91,8 +91,10 @@ options.registry.SnippetPopup = options.Class.extend({
      * @see this.selectClass for parameters
      */
     moveBlock: function (previewMode, widgetValue, params) {
-        const $container = $(widgetValue === 'moveToFooter' ? 'footer' : 'main');
-        this.$target.closest('.s_popup').prependTo($container.find('.oe_structure:o_editable').first());
```

Comment:
You are right. A better way of doing would be to write : 
```js
const containerEl = this.$target[0].ownerDocument.querySelector(widgetValue === 'moveToFooter' ? 'footer' : 'main');
const whereEl = containerEl.querySelector('.oe_structure.o_editable, .o_editable .oe_structure');
const popupEl = this.$target[0].closest('.s_popup');
whereEl.prepend(popupEl);
```
However, even this way of writing is not perfect as a case like this could occur :
<div o_editable
…
    <div o_not_editable
…
          <div oe_structure

This case, would be returned by the ```querySelector``` but it should not.
However, after investigation, it appears that the class ```o_not_editable``` is never added dynamically by a JavaScript code. Nonetheless, it could be added by a potential client customization so, from my point of view, it should be better to use the override of jQuery and use:
```js
const $container = this.$target[0].ownerDocument.defaultView.$(widgetValue === 'moveToFooter' ? 'footer' : 'main');
this.$target.closest('.s_popup').prependTo($container.find('.oe_structure:o_editable').first());
```